### PR TITLE
Added references to network watcher examples + test for delete connec…

### PR DESCRIPTION
**Pull request by irrogozh**
…tion monitor

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2017-11-01/networkWatcher.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
           {
             "name": "parameters",
             "in": "body",
-            "required": true,
+            "required": false,
```
Comment (fearthecowboy) [inline specification/network/resource-manager/Microsoft.Network/stable/2017-11-01/networkWatcher.json]: This seems like a significant change; why are the parameters no longer required?

**Comment by fearthecowboy** [comment]
I'm going to try and see what's not cooperating here...

**Comment by fearthecowboy** [comment]
I changed the `NetworkWatcherCreate.json` example to have parameters like this:

``` json
"parameters": {                           
  "api-version": "2017-11-01",            
  "subscriptionId": "subid",              
  "resourceGroupName": "rg1",             
  "networkWatcherName": "nw1",            
  "parameters": {                         
    "location": "eastus"                  
  }                                       
},                                        
```

And it fixed the problem with `location` 

After that you still have more example validation errors to fix up.

You can run the model validator by itself to see the output:

``` powershell
C:\irrogozh\specification\network\resource-manager > npm install -g oav
C:\irrogozh\specification\network\resource-manager > oav validate-example .\Microsoft.Network\stable\2017-11-01\networkWatcher.json

```

Which currently shows me: 

``` javascript
error: Found errors in validating the response with statusCode "200" for x-ms-example "Get security group view" in operation "NetworkWatchers_GetVMSecurityRules".:

{ code: 'RESPONSE_VALIDATION_ERROR',
  id: 'OAV108',
  message: 'Found errors in validating the response with statusCode "200" for x-ms-example "Get security group view" in operation "NetworkWatchers_GetVMSecurityRules".',
  innerErrors:
   [ { code: 'INVALID_RESPONSE_BODY',
       errors:
        [ { code: 'ONE_OF_MISSING',
            params: [],
            message: 'Data does not match any schemas from \'oneOf\'',
            path: [ 'networkInterfaces' ],
            inner:
             [ { code: 'ONE_OF_MISSING',
                 params: [],
                 message: 'Data does not match any schemas from \'oneOf\'',
                 path: [ 'networkInterfaces', '0' ],
                 inner:
                  [ { code: 'ONE_OF_MISSING',
                      params: [],
                      message: 'Data does not match any schemas from \'oneOf\'',
                      path: [ '0', 'securityRuleAssociations' ],
                      inner:
                       [ { code: 'ONE_OF_MISSING',
                           params: [],
                           message: 'Data does not match any schemas from \'oneOf\'',
                           path: [ 'securityRuleAssociations', 'networkInterfaceAssociation' ],
                           inner:
                            [ { code: 'ONE_OF_MISSING',
                                params: [],
                                message: 'Data does not match any schemas from \'oneOf\'',
                                path: [ 'networkInterfaceAssociation', 'securityRules' ],
                                inner:
                                 [ { code: 'ONE_OF_MISSING',
                                     params: [],
                                     message: 'Data does not match any schemas from \'oneOf\'',
                                     path: [ 'securityRules', '0' ],
                                     inner:
                                      [ { code: 'ONE_OF_MISSING',
                                          params: [],
                                          message: 'Data does not match any schemas from \'oneOf\'',
                                          path: [ '0', 'properties' ],
                                          inner:
                                           [ { code: 'ENUM_MISMATCH',
                                               params: [ 'TCP' ],
                                               message: 'No enum match for: TCP',
                                               path: [ 'properties', 'protocol' ],
                                               description: 'Network protocol this rule applies to. Possible values are \'Tcp\', \'Udp\', and \'*\'.' },
                                             { code: 'INVALID_TYPE',
                                               params: [ 'null', 'object' ],
                                               message: 'Expected type null but found type object',
                                               path: [ 'properties' ] } ] },
                                        { code: 'INVALID_TYPE',
                                          params: [ 'null', 'object' ],
                                          message: 'Expected type null but found type object',
                                          path: [ '0' ] } ] },
                                   { code: 'INVALID_TYPE',
                                     params: [ 'null', 'array' ],
                                     message: 'Expected type null but found type array',
                                     path: [ 'securityRules' ] } ] },
                              { code: 'INVALID_TYPE',
                                params: [ 'null', 'object' ],
                                message: 'Expected type null but found type object',
                                path: [ 'networkInterfaceAssociation' ] } ] },
                         { code: 'INVALID_TYPE',
                           params: [ 'null', 'object' ],
                           message: 'Expected type null but found type object',
                           path: [ 'securityRuleAssociations' ] } ] },
                    { code: 'INVALID_TYPE',
                      params: [ 'null', 'object' ],
                      message: 'Expected type null but found type object',
                      path: [ '0' ] } ] },
               { code: 'INVALID_TYPE',
                 params: [ 'null', 'array' ],
                 message: 'Expected type null but found type array',
                 path: [ 'networkInterfaces' ] } ] } ],
       message: 'Invalid body: Data does not match any schemas from \'oneOf\'',
       path: [] } ] }
error: Found errors in validating the response with statusCode "202" for x-ms-example "Get security group view" in operation "NetworkWatchers_GetVMSecurityRules".:

{ code: 'RESPONSE_VALIDATION_ERROR',
  id: 'OAV108',
  message: 'Found errors in validating the response with statusCode "202" for x-ms-example "Get security group view" in operation "NetworkWatchers_GetVMSecurityRules".',
  innerErrors:
   [ { code: 'INVALID_RESPONSE_BODY',
       errors:
        [ { code: 'ONE_OF_MISSING',
            params: [],
            message: 'Data does not match any schemas from \'oneOf\'',
            path: [ 'networkInterfaces' ],
            inner:
             [ { code: 'ONE_OF_MISSING',
                 params: [],
                 message: 'Data does not match any schemas from \'oneOf\'',
                 path: [ 'networkInterfaces', '0' ],
                 inner:
                  [ { code: 'ONE_OF_MISSING',
                      params: [],
                      message: 'Data does not match any schemas from \'oneOf\'',
                      path: [ '0', 'securityRuleAssociations' ],
                      inner:
                       [ { code: 'ONE_OF_MISSING',
                           params: [],
                           message: 'Data does not match any schemas from \'oneOf\'',
                           path: [ 'securityRuleAssociations', 'networkInterfaceAssociation' ],
                           inner:
                            [ { code: 'ONE_OF_MISSING',
                                params: [],
                                message: 'Data does not match any schemas from \'oneOf\'',
                                path: [ 'networkInterfaceAssociation', 'securityRules' ],
                                inner:
                                 [ { code: 'ONE_OF_MISSING',
                                     params: [],
                                     message: 'Data does not match any schemas from \'oneOf\'',
                                     path: [ 'securityRules', '0' ],
                                     inner:
                                      [ { code: 'ONE_OF_MISSING',
                                          params: [],
                                          message: 'Data does not match any schemas from \'oneOf\'',
                                          path: [ '0', 'properties' ],
                                          inner:
                                           [ { code: 'ENUM_MISMATCH',
                                               params: [ 'TCP' ],
                                               message: 'No enum match for: TCP',
                                               path: [ 'properties', 'protocol' ],
                                               description: 'Network protocol this rule applies to. Possible values are \'Tcp\', \'Udp\', and \'*\'.' },
                                             { code: 'INVALID_TYPE',
                                               params: [ 'null', 'object' ],
                                               message: 'Expected type null but found type object',
                                               path: [ 'properties' ] } ] },
                                        { code: 'INVALID_TYPE',
                                          params: [ 'null', 'object' ],
                                          message: 'Expected type null but found type object',
                                          path: [ '0' ] } ] },
                                   { code: 'INVALID_TYPE',
                                     params: [ 'null', 'array' ],
                                     message: 'Expected type null but found type array',
                                     path: [ 'securityRules' ] } ] },
                              { code: 'INVALID_TYPE',
                                params: [ 'null', 'object' ],
                                message: 'Expected type null but found type object',
                                path: [ 'networkInterfaceAssociation' ] } ] },
                         { code: 'INVALID_TYPE',
                           params: [ 'null', 'object' ],
                           message: 'Expected type null but found type object',
                           path: [ 'securityRuleAssociations' ] } ] },
                    { code: 'INVALID_TYPE',
                      params: [ 'null', 'object' ],
                      message: 'Expected type null but found type object',
                      path: [ '0' ] } ] },
               { code: 'INVALID_TYPE',
                 params: [ 'null', 'array' ],
                 message: 'Expected type null but found type array',
                 path: [ 'networkInterfaces' ] } ] } ],
       message: 'Invalid body: Data does not match any schemas from \'oneOf\'',
       path: [] } ] }
```

**Review by fearthecowboy — APPROVED** [review]
Ok, I think this is good. At least it doesn't have new validation problems that are related to this PR.

**Comment by fearthecowboy** [comment]
@irrogozh - Are you OK to merge this now?

**Comment by irrogozh** [comment]
Of cause, Garrett, let's merge it :)

**Pull request merged**